python/src/space/core/storage.py

```diff
@@ -493,7 +563,7 @@ def commit(self, patch: Optional[rt.Patch]) -> None:
       self._result = JobResult(JobResult.State.SKIPPED)
       return
 
-    self._storage.commit(patch)
+    self._storage.commit(patch, self._branch)
```

Comment:
_branch is always storage.current_branch, can we remove the second parameter and get its value inside commit?

Reply:
_branch is always the value storage.current_branch of entering the transaction, but it could be changed after that and before commit,so we will need to remember it here so that we won't commit the patch to a wrong branch. see test_concurrent_write_to_different_branch